dhall/src/Dhall/Parser/Expression.hs

```diff
@@ -361,7 +350,10 @@ parsers embedded = Parsers {..}
         , BoolNE                  <$ _notEqual     <* whitespace
         ]
 
-    applicationExpression = do
+    applicationExpression = snd <$> applicationExpression'
+
+    applicationExpression' :: Parser (ApplicationExprType, Expr Src a)
```

Comment:
Perhaps call this `applicationExpressionWithInfo` or something like that.  I try to avoid using `'` in names